src/SearchResult.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+
+export default function SearchResult({ item, checkDiff, showProfile }) {
+  const [select, setSelect] = useState(false);
+
+  function changeSelected() {
+    setSelect(!select);
+  }
```

Comment:
breaking down each row into its own `SearchResult` component works really well, this means you can easily highlight each row, due to each row having its own state